test/srt/hicache/test_hicache_storage_mooncake_backend.py

```diff
@@ -0,0 +1,297 @@
+                    # and it's been a few seconds since startup
+                    if (
+                        time.time() - start_time > 5
+                    ):  # Give master service time to initialize
+                        master_ready = True
+                        print("Mooncake master service is ready")
+
+                # Both services should be ready
+                if metadata_ready and master_ready:
+                    services_ready = True
+                    print("All Mooncake services are ready")
+                    break
+
+            except Exception as e:
+                print(f"Error checking service readiness: {e}")
+
+            time.sleep(2)
+
+        if not services_ready:
+            print(
+                "Warning: Mooncake services may not be fully ready, continuing anyway..."
+            )
+
+        return services_ready
+
+    @classmethod
+    def _stop_mooncake_services(cls):
+        """Stop Mooncake services"""
+        print("Stopping Mooncake services...")
+
+        # Stop metadata service
+        if hasattr(cls, "metadata_service_process") and cls.metadata_service_process:
+            try:
+                os.killpg(os.getpgid(cls.metadata_service_process.pid), 9)
+                cls.metadata_service_process.wait(timeout=5)
+                print("Mooncake metadata service stopped")
+            except (ProcessLookupError, subprocess.TimeoutExpired, OSError) as e:
+                print(f"Warning: Could not stop Mooncake metadata service: {e}")
+
+        # Stop master service
+        if hasattr(cls, "master_service_process") and cls.master_service_process:
+            try:
+                os.killpg(os.getpgid(cls.master_service_process.pid), 9)
+                cls.master_service_process.wait(timeout=5)
+                print("Mooncake master service stopped")
+            except (ProcessLookupError, subprocess.TimeoutExpired, OSError) as e:
+                print(f"Warning: Could not stop Mooncake master service: {e}")
+
+    @classmethod
+    def _get_additional_server_args_and_env(cls):
+        """Get additional server arguments specific to configuration - override in subclasses"""
+
+        server_args = {
+            "--tp-size": 1,
+            "--hicache-ratio": 1.2,
+            "--hicache-storage-backend": "mooncake",
+        }
+
+        # Set the environment variables for Mooncake using dynamic ports
+        env_vars = {
+            "MOONCAKE_MASTER": f"127.0.0.1:{cls.mooncake_master_port}",
+            "MOONCAKE_PROTOCOL": "rdma",
+            "MOONCAKE_DEVICE": "",
+            "MOONCAKE_TE_META_DATA_SERVER": f"http://127.0.0.1:{cls.mooncake_metadata_port}/metadata",
+            "MOONCAKE_GLOBAL_SEGMENT_SIZE": "4294967296",  # 4 GiB
+        }
+
+        return server_args, env_vars
+
+
+# Same as #10131, layer first layout test
+class TestMooncakeBackendLayerFirstLayout(
```

Comment:
It seems like the Mooncake backend does not support layer-first layout yet. Perhaps we could test MHA and MLA, respectively, as they have quite different processing logic in the Mooncake backend.